pkg/controller/kubedirectorcluster/cluster.go

```diff
@@ -206,9 +206,57 @@ func (r *ReconcileKubeDirectorCluster) syncCluster(
 				shared.EventReasonCluster,
 				"stable",
 			)
+
+			amIBeingConnectedToThis := func(otherCluster kdv1.KubeDirectorCluster) bool {
+				for _, connectedName := range otherCluster.Spec.Connections.Clusters {
+					if cr.Name == connectedName {
+						return true
+					}
+				}
+				return false
+			}
+
+			// Once the cluster is deemed ready, if this cluster is connected to any cluster then
+			// we need to notify that cluster that configmeta here has
+			// changed, so bump up connectionsGenerationToProcess for that cluster
+			allClusters := &kdv1.KubeDirectorClusterList{}
+			shared.List(context.TODO(), allClusters)
+			// notify clusters to which this cluster is
+			// connected
+			for _, kubecluster := range allClusters.Items {
+				if amIBeingConnectedToThis(kubecluster) {
+					shared.LogInfof(
+						reqLogger,
+						cr,
+						shared.EventReasonConfigMap,
+						"cluster {%s} is a connection to another cluster, updating {%s}'s configmeta",
+						cr.Name,
+						kubecluster.Name,
+					)
+					updateMetaGenerator := &kubecluster
+					updateMetaGenerator.Spec.ConnectionsGenToProcess = kubecluster.Spec.ConnectionsGenToProcess + 1
+					//Notify cluster by incrementing configmetaGenerator
+					wait := time.Second
+					maxWait := 4096 * time.Second
+					for {
```

Comment:
The process of
* read cluster object
* increment the observed gen number in spec
* write cluster object

Needs to be inside this for loop. That's because the most likely reason for this update to fail is because of a conflict with some other write... so, we will need to read the cluster object again to get a copy of the object that includes the new updated resource version.

Also if we want to guarantee that we actually are bumping the spec gen by 1, then that's another reason to have the whole read-modify-write process inside the loop here.

As a side note: you COULD sidestep this looping by using PATCH, now that we have moved to a new version of the SDK that supports PATCH. However if you don't want to dig into that right now I don't blame you. I think it's more straightforward at the moment just to move the read-modify-write inside the for loop.